### ISSUE TITLE
Empty/blank bundle preview and 500 internal server error for diagram.svg

### ISSUE THREAD

**Issue by sparkiegeek** [canonical/jujucharms.com#70]
https://jujucharms.com/u/landscape/landscape-dense-maas/10 has an empty grey canvas for the preview section.

Looking in the console I see a 500 error for diagram.svg, manually visiting the [URL](https://api.jujucharms.com/charmstore/v4/~landscape/bundle/landscape-dense-maas-10/diagram.svg) gives me

`{"Message":"cannot create canvas: service \"apache2\" does not have a valid position","Code":""}`

**Comment by mitechie** [comment]
The issue here is the bundle doesn't have x/y annotations for the jujusvg tool to lay out the bundle and visualize it. We've got an issue on jujusvg to add a default layout algorithm but it's not complete yet.

https://github.com/juju/jujusvg/issues/15

Adding x/y annotations to the bundle will help things get generated correctly. See the mongodb-cluster bundle for example:

https://api.jujucharms.com/charmstore/v4/bundle/mongodb-cluster-4/archive/bundle.yaml

Now this bundle has issue exporting from the gui because it's using colocation which we're working to add support to the deployer and then the gui will be updated to support exporting a format with colocation (lxc placement) that the deployer will understand. Until then, you'll have to add annotation somewhat manually. I suggested to another user they recreate the bundle in the demo.jujucharms.com website without it being dense, and just laying it out, and then exporting the yaml and copy/pasting the x/y annotations to the dense bundle to help it lay out correctly.
